gtfs_rt_validator_api.py

```diff
@@ -218,13 +293,36 @@ def download_rt_files(dst_dir, fs=None, date="2021-08-01", glob_path=None):
 
     print(f"Copying {len(to_copy)} files")
 
-    with futures.ThreadPoolExecutor(max_workers=N_THREAD_WORKERS) as pool:
-        list(pool.map(lambda args: fs.get(*args), to_copy))
+    src_files, dst_files = zip(*to_copy)
+    fs.get(list(src_files), list(dst_files))
 
 
 # Rectangling =================================================================
 
 
+def rollup_error_counts(rt_dir):
+    result_files = Path(rt_dir).glob("*.results.json")
+
+    code_counts = []
+    for path in result_files:
+        metadata = parse_pb_name_data(path)
+
+        result_json = json.load(path.open())
+        for entry in result_json:
+            code_counts.append({
+                "calitp_itp_id": metadata["itp_id"],
+                "calitp_url_number": metadata["url_number"],
+                "calitp_extracted_at": metadata["dt"],
+                "rt_feed_type": metadata["src_fname"],
+                "rule_id": entry["errorMessage"]["validationRule"]["errorId"],
```

Comment:
Let's stay consistent and call this `error_id`.

```suggestion
                "error_id": entry["errorMessage"]["validationRule"]["errorId"],
```

Reply:
Adding some context on how rule seems to align with the validator docs / outputs, but am happy to change to `error_id`, if useful! I was caught a bit off-guard by this behavior in the validator..

AFAICT these are weirdly not errors, but what the GTFS RT validator refers to as "rules" (and the GTFS validator calls "codes"):

![image](https://user-images.githubusercontent.com/2574498/146815509-5bf36595-fd90-4966-b162-e8324b21c1d1.png)

https://github.com/CUTR-at-USF/gtfs-realtime-validator#rules